src/storage/table/column_segment.cpp

```diff
@@ -91,7 +91,12 @@ void ColumnSegment::Scan(ColumnScanState &state, idx_t scan_count, Vector &resul
 	} else {
 		D_ASSERT(result.GetVectorType() == VectorType::FLAT_VECTOR);
 		ScanPartial(state, scan_count, result, result_offset);
-		D_ASSERT(result.GetVectorType() == VectorType::FLAT_VECTOR);
+#ifdef DEBUG
+		// If we are scanning an entire Vector we don't really care about the resulting type
+		if (scan_count != STANDARD_VECTOR_SIZE) {
```

Comment:
Isn't this what the `entire_vector` flag is for?

Reply:
Yep, missed this when I was first figuring out the issue